File: MusicPlayer.Core/MusicTrack.cs
using MediaPlayer.Core;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace MusicPlayer.Core
{
  public class MusicTrack : PlayableMedia
  {
    [Required, StringLength(128)]
    public string Title { get; set; }
    [Required]
    [DefaultValue("Unknown")]
    public string Album { get; set; }
    [DefaultValue("Unknown")]
    public string Artist { get; set; }
    [Required]
    public int Duration { get; set; }
    [Required]
    public string FilePath { get; set; }
    public int Notation { get; set; }

    public int SampleRate { get; set; }

    public int Bitrate { get; set; }
    public int Channels { get; set; }
    public uint DiscNumber { get; set; }
    public uint TrackNumber { get; set; }
    public override string Source { get => FilePath; }
    }
}

File: MusicPlayer/ClientApp/src/store/Entities/MusicTrack.ts
import BaseMediaEntity from "./BaseMediaEntity";

export default interface MusicTrack extends BaseMediaEntity
{
  title: string;
  sorttitle: string;
  album: string;
  artist: string;
  duration: number;
  notation: number;
  samplerate: number;
  bitrate: number;
  channels: number;
  discnumber: number;
  tracknumber: number;
  filepath: string;
}